docs/audits.md

```diff
@@ -1317,6 +1317,37 @@ Address the source of obfuscation by simplifying the expression,
                   repository: octocat/hello-world
         ```
 
+## `stale-action-refs`
+
+| Type     | Examples                | Introduced in | Works offline  | Enabled by default | Configurable |
+|----------|-------------------------|---------------|----------------|--------------------|--------------|
+| Workflow, Action  | N/A            | v1.7.0        | ❌            | ✅                | ❌          |
+
+Checks for `#!yaml uses:` clauses which pin an action using a SHA reference,
+but where that reference does not point to a Git tag.
+
+When using an action commit which is not a Git tag / release version, that commit
+might contain bugs or vulnerabilities which have not been publicly documented
+because they might have been fixed before the subsequent release. Additionally,
+because changelogs are usually only published for releases, it is difficult to
+tell which changes of the subsequent release the pinned commit includes.
+
+!!! note
+
+    This is a `--pedantic` only audit because the detected situation is not
+    a vulnerability per se. But it might be worth investigating which commit
+    the SHA reference points to, and why not a SHA reference pointing to a
+    Git tag is used.
+
+    Some action repositories use a "rolling release branch" strategy where
+    all commits on a certain branch are considered releases. In such a case
+    findings of this audit can likely be ignored.
```

Comment:
I was thinking a bit, and I think we _could_ detect this scenario by looking up the SHA ref, and then seeing whether it belongs to a branch other that the default one. But that's likely to expensive in terms of API calls and slow, so I'm OK with not doing it for now.

Reply:
It could also lead to false negatives when that branch is not a rolling release branch, but instead some work-in-progress branch containing bugs.

Reply:
True. There's probably no good general solution here, then.